### ISSUE TITLE
Doesn't compile on thumbv6m-none-eabi

### ISSUE THREAD

**Issue by chxry** [rhaiscript/rhai#903]
Building latest git version and v1.19.0 fails on thumbv6m-none-eabi with features=["no_std"]. 
All the errors are from spin using AtomicUsize operations.
```
error[E0599]: no method named `fetch_add` found for struct `AtomicUsize` in the current scope
   --> /home/chxry/.cargo/registry/src/index.crates.io-6f17d22bba15001f/spin-0.7.1/src/rw_lock.rs:197:31
    |
197 |         let value = self.lock.fetch_add(READER, Ordering::Acquire);
    |                               ^^^^^^^^^ method not found in `AtomicUsize
```

**Comment by schungx** [comment]
Maybe spin locking is not supported for that CPU...

**Comment by schungx** [comment]
A bit searching reveals that this problem is quite common on the v6m...

https://github.com/tokio-rs/bytes/issues/461
https://users.rust-lang.org/t/issue-with-atomicbool-when-trying-to-build-project-for-v6m/59729
https://github.com/fereidani/rclite/issues/7

**Comment by schungx** [comment]
Wonder if you have any success?

**Comment by chxry** [comment]
For me doing [this](https://github.com/chxry/rhai/commit/af3647c6f48b29585f43b2bd0f0aa14faec29d6e) worked, however I haven't tested how this affects other systems.

**Comment by schungx** [comment]
This looks interesting.  I was not aware of the `compat_sync` feature.  Can you make a PR?

**Comment by chxry** [comment]
Fixed by #919
